python/lsst/meas/algorithms/detection.py

```diff
@@ -392,7 +408,7 @@ def calculateKernelSize(self, sigma):
         return (int(sigma * self.config.nSigmaForKernel + 0.5)//2)*2 + 1  # make sure it is odd
 
     def getPsf(self, exposure, sigma=None):
-        """Retrieve the PSF for an exposure
+        """Generate or retrieve a PSF for an exposure.
```

Comment:
I find the use of "generate" potentially a little confusing. I assume this means that if one is premade and lives on the object then it is returned, and if something behind it needs some sort of making, then that is done, right? But will a new user think like that, or might that suggest some sort of fitting to "generate" the PSF? Just thinking aloud though, so feel free to do as you like.

Reply:
Thanks for asking about this: I dug into it more and it left me with some questions that I've asked on #dm-science-pipelines. I've reworded to better reflect what the code actually does right now.